python/src/main/python/drivers/runspv.py

```diff
@@ -925,13 +958,159 @@ def amberscriptify_image(
 
     result += '[test]\n'
     result += '## Uniforms\n'
-    result += uniform_json_to_amberscript(uniform_json)
+    result += uniform_json_to_vkscript(uniform_json)
     result += '\n'
     result += 'draw rect -1 -1 2 2\n'
 
     return result
 
 
+def amberscript_uniform_buffer_decl(uniform_json):
+    '''
+    Returns the string representing AmberScript version of uniform declarations.
+    Skips the special '$compute' key, if present.
+
+    {
+      "myuniform": {
+        "func": "glUniform1f",
+        "args": [ 42.0 ],
+        "binding": 3
+      },
+      "$compute": { ... will be ignored ... }
+    }
+
+    becomes:
+
+    # myuniform
+    BUFFER myuniform DATA_TYPE float DATA
+      42.0
+    END
+
+    '''
+
+    UNIFORM_TYPE = {
+        'glUniform1i': 'int32',
+        'glUniform2i': 'vec2<int32>',
+        'glUniform3i': 'vec3<int32>',
+        'glUniform4i': 'vec4<int32>',
+        'glUniform1f': 'float',
+        'glUniform2f': 'vec2<float>',
+        'glUniform3f': 'vec3<float>',
+        'glUniform4f': 'vec4<float>',
+    }
+
+    result = ''
+    with open(uniform_json, 'r') as f:
+        j = json.load(f)
+    for name, entry in j.items():
+
+        if name == '$compute':
+            continue
+
+        func = entry['func']
+        if func not in UNIFORM_TYPE.keys():
```

Comment:
We should probably change the following to "raise"; @paulthomson what do you think?